File: Safari-609.1.11.1/LayoutTests/http/tests/workers/service/resources/cacheable-script-worker.php
<?php
    $max_age = 12 * 31 * 24 * 60 * 60; //one year
    header('Cache-Control: public, max-age=' . $max_age);
    header('Content-Type: text/javascript');

    $randomId = '';
    $charset = 'ABCDEFGHIKLMNOPQRSTUVWXYZ0123456789';
    for ($i = 0; $i < 16; $i++)
        $randomId .= $charset[rand(0, strlen($charset) - 1)];

    echo 'let randomId = "' . $randomId . '";';
?>

self.addEventListener("message", function(e) {
   e.source.postMessage(randomId);
});

File: LayoutTests/http/tests/workers/service/resources/cacheable-script-worker.py
#!/usr/bin/env python3

import sys
from random import randint

max_age = 12 * 31 * 24 * 60 * 60

sys.stdout.write(
    'Cache-Control: public, max-age={}\r\n'
    'Content-Type: text/javascript\r\n\r\n'.format(max_age)
)

randomId = ''
charset = 'ABCDEFGHIKLMNOPQRSTUVWXYZ0123456789'
for _ in range(0, 16):
    randomId += charset[randint(0, len(charset) - 1)]

sys.stdout.write(
    'let randomId = "{}";\n'
    'self.addEventListener("message", function(e) {{\n'
    '   e.source.postMessage(randomId);\n'
    '}});\n'.format(randomId)
)